### ISSUE TITLE
Factory for interfaces

### ISSUE THREAD

**Issue by picrap** [ArxOne/MrAdvice#8]
Allow to create interfaces implementation using one or many aspects at type level.
Ideas:
```csharp
// may take any arbitrary number of aspects
var i = Weavisor.Factory.Create<Interface,MyAdvice>();
// or
// takes only one aspect but is much nicer
var j = MyAdvice.Handle<Interface>();
```

**Comment by picrap** [comment]
This is probably going to be something like `MyAdvice.Handle<Interface>()`, because it does not make sense to inject multiple advices there (an interface is a contract, not an implementation, and aspects go in the implementations).

**Comment by picrap** [comment]
Second thought: not sure this is a job for Weavisor (a transparent proxy is enough here).

**Comment by picrap** [comment]
Or maybe a simple wrapper using RealProxy?

**Comment by picrap** [comment]
RealProxy is not available in PCL. Switching back to original idea (build time weaving).

**Comment by picrap** [comment]
Implemented in 0.3.8
